_source/_posts/2023-01-23-keycloak-kubernetes-prod.md

```diff
@@ -0,0 +1,637 @@
+
+### Create an Auth0 account
+
+Sign up at [Auth0](https://auth0.com/signup) and install the [Auth0 CLI](https://github.com/auth0/auth0-cli). Then run:
+
+```shell
+auth0 login
+```
+
+The terminal will display a device confirmation code and open a browser session to activate the device.
+
+**NOTE**: My browser was not displaying anything, so I had to manually activate the device by opening the URL `https://auth0.auth0.com/activate?user_code={deviceCode}`.
+
+On successful login, you will see the tenant, which you will use as issuer later:
+
+```
+✪ Welcome to the Auth0 CLI 🎊
+
+If you don't have an account, please create one here: https://auth0.com/signup.
+
+Your device confirmation code is: KGFL-LNVB
+
+ ▸    Press Enter to open the browser to log in or ^C to quit...
+
+Waiting for the login to complete in the browser... ⣻Opening in existing browser session.
+Waiting for the login to complete in the browser... done
+
+ ▸    Successfully logged in.
+ ▸    Tenant: dev-avup2laz.us.auth0.com
+```
+
+The next step is to create a client app:
+
+```shell
+auth0 apps create
+```
+
+When prompted, choose the following options:
+
+- Name: **jhipster**
+- Description: **<optional description>**
+- Type: **Regular Web Application**
+- Callback URLs: **https://keycloak.\<namespace\>.\<public-ip\>.nip.io/login/oauth2/code/auth0**
+- Logout URLs: **http://gateway.\<namespace\>.\<public-ip\>.nip.io**
+
+Once the app is created, you will see the OIDC app's configuration:
+
+```
+Name: jhipster
+Description: jhipster microservices
+Callback URLs: ***
+Allowed Logout URLs: ***
+
+=== dev-avup2laz.us.auth0.com application created
+
+  CLIENT ID            ***
+  NAME                 jhipster
+  DESCRIPTION          jhipster microservices
+  TYPE                 Regular Web Application
+  CALLBACKS            ***
+  ALLOWED LOGOUT URLS  ***
+  ALLOWED ORIGINS
+  ALLOWED WEB ORIGINS
+  TOKEN ENDPOINT AUTH
+  GRANTS               implicit, authorization_code, refresh_token, client_credentials
+
+ ▸    Quickstarts: https://auth0.com/docs/quickstart/webapp
+ ▸    Hint: Emulate this app's login flow by running `auth0 test login ***`
+ ▸    Hint: Consider running `auth0 quickstarts download ***`
+```
+
+**NOTE**: The client secret is [not displayed](https://github.com/auth0/auth0-cli/issues/488) in the CLI output for regular applications. You can run `auth0 apps open` to open a browser in the application configuration. Then copy the secret from the **Settings** tab. In the example above, the tenant is `dev-avup2laz.us.auth0.com`.
```

Comment:
If you use `--reveal-secrets` as I did above, you can remove this line.

Reply:
Updated